test/ng/directive/ngOptionsSpec.js

```diff
@@ -922,6 +922,25 @@ describe('ngOptions', function() {
       expect(scope.selected).toEqual([20]);
       expect(element).toEqualSelectValue([20], true);
     });
+
+
+    it('should ignore array values beginning with $$', function() {
+      scope.arr = ['one', 'two', '$$hashKey'];
```

Comment:
It makes sense to ignore properties of an array beginning with "$$", but for elements I'm not really convinced